src/Psalm/Internal/Analyzer/Statements/Block/ForeachAnalyzer.php

```diff
@@ -93,6 +93,10 @@ public static function analyze(
 
         if ($stmt->valueVar instanceof PhpParser\Node\Expr\Variable && is_string($stmt->valueVar->name)) {
             $safe_var_ids['$' . $stmt->valueVar->name] = true;
+            $statements_analyzer->foreach_var_locations['$' . strtolower($stmt->valueVar->name)][] = new CodeLocation(
```

Comment:
why are you normalising the variable name? Variables in PHP are case-sensitive.

Reply:
Good point.

Reply:
Fixed.